src/components/LazyImage.js

```diff
@@ -0,0 +1,18 @@
+import styles from "./lazyImage.module.scss";
+import { useEffect, useRef } from "react";
+
+export default function LazyImage({ src, alt }) {
+  const imageRef = useRef(null);
+  useEffect(() => {
+    const io = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          const image = entry.target;
+          image.src = src;
+        }
+      });
```

Comment:
The IntersectionObserver get 2 parameters. Please pass to the second one the margin, so you images will start loading before users will see them